### PR TITLE
Create IUnknownStrategy for GlobalInterfaceTable

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Create a Global Interface Table IUnknownStrategy so we can use the same machinery the runtime uses to expose COM interfaces as objects with AgileComPointer-like support.

Change all of the direct Marshal calls to use a ComHelpers wrapper method.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9845)

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
@jkoritzinsky here we're trying to leverage the stuff we talked about

File: src/System.Windows.Forms/src/System/Windows/Forms/Application.cs [hunk 1]
```diff
@@ -143,43 +139,6 @@ public static RegistryKey CommonAppDataRegistry
     internal static string CommonAppDataRegistryKeyName
         => $"Software\\{CompanyName}\\{ProductName}\\{ProductVersion}";
 
-    internal static bool UseEverettThreadAffinity
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/Application.cs:146]: Why was this removed?

**Review by lonitra — APPROVED** [review]
Just a question, otherwise LGTM

File: src/System.Windows.Forms/src/System/Windows/Forms/Application.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/Application.cs:146]: It is never called. :) Forgot to mention it.

**Review by jkoritzinsky — APPROVED** [review]
Love to see new usages of StrategyBasedComWrappers!
